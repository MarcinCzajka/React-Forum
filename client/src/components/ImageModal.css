.modalImageContainer {
    width: auto;
    height: auto;
    background-color: #fff;
    border-radius: 15px;
    padding:15px;
}

.modalImage {
    
    max-width: 90vw;
    max-height: 80vh;
}