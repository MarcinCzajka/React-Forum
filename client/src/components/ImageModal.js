import React from 'react';
import { Dimmer, Icon } from "semantic-ui-react";
import './ImageModal.css'

class ImageModal extends React.Component {
    constructor(props) {
        super(props);

        this.state = {image: this.props.image, open: false};
    }

    open = () => {
        this.setState({open: true});
    }

    close = () => {
        this.setState({open: false});
    }

    render() {
        const { open } = this.state
        
        return (
            <Dimmer page onClickOutside={this.close} active={open} >
                <div className='modalImageContainer' >
                    <img className='modalImage' src={this.state.image} />
                    <Icon name='times' onClick={this.close} className='modalCloseIcon' />
                </div>
            </Dimmer>
        )
    }
}

export default ImageModal