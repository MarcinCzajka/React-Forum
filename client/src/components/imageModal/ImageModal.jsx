import React from 'react';
import PropTypes from 'prop-types';
import { Dimmer, Icon, Loader } from "semantic-ui-react";
import './ImageModal.css'

class ImageModal extends React.Component {
    constructor(props) {
        super(props);

        this.state = {open: false, loading: true};
    }

    open = () => {
        this.setState({open: true});
    }

    close = () => {
        this.setState({open: false, loading: true});
    }

    hideLoader = () => {
        this.setState({loading: false})
    }

    render() {
        const { open, loading } = this.state
        
        return (
            <Dimmer page onClickOutside={this.close} active={open} >
                <div className='modalImageContainer' >
                    <Dimmer inverted active={loading}>
                        <Loader size='big' />
                    </Dimmer>
                    <img className='modalImage' onLoad={this.hideLoader} alt={this.props.alt} src={this.props.image} />
                    <Icon name='times' onClick={this.close} className='modalCloseIcon' />
                </div>
            </Dimmer>
        )
    }
}

ImageModal.propTypes = {
    image: PropTypes.string,
    alt: PropTypes.string
}

export default ImageModal