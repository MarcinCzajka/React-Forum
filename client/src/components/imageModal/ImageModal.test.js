import React from 'react';
import { create } from "react-test-renderer";
import { shallow } from 'enzyme';
import ImageModal from './ImageModal';

jest.mock('semantic-ui-react/dist/commonjs/addons/Portal/Portal', () => ({ children }) => children);

const redDotBase64 = `data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAUA
AAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO
    9TXL0Y4OHwAAAABJRU5ErkJggg==`;

describe('ImageModal component', () => {
    test('renders without crashing', () => {
        shallow(<ImageModal />);
    });

    test('match snapshot w/o image source', () => {
        const component = create(<ImageModal />);
        expect(component).toMatchSnapshot();
    });

    test('match snapshot /w image source', () => {
        const component = create(<ImageModal image={redDotBase64} alt="sample alt text" />);
        expect(component).toMatchSnapshot();
    });

});