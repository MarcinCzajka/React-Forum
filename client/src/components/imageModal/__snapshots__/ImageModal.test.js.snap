// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageModal component match snapshot /w image source 1`] = `
<div
  className="ui page dimmer"
  onClick={[Function]}
>
  <div
    className="content"
  >
    <div
      className="modalImageContainer"
    >
      <div
        className="ui active transition visible inverted dimmer"
        onClick={[Function]}
      >
        <div
          className="content"
        >
          <div
            className="ui big loader"
          />
        </div>
      </div>
      <img
        alt="sample alt text"
        className="modalImage"
        onLoad={[Function]}
        src="data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAUA
AAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO
    9TXL0Y4OHwAAAABJRU5ErkJggg=="
      />
      <i
        aria-hidden="true"
        className="times icon modalCloseIcon"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`ImageModal component match snapshot w/o image source 1`] = `
<div
  className="ui page dimmer"
  onClick={[Function]}
>
  <div
    className="content"
  >
    <div
      className="modalImageContainer"
    >
      <div
        className="ui active transition visible inverted dimmer"
        onClick={[Function]}
      >
        <div
          className="content"
        >
          <div
            className="ui big loader"
          />
        </div>
      </div>
      <img
        className="modalImage"
        onLoad={[Function]}
      />
      <i
        aria-hidden="true"
        className="times icon modalCloseIcon"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;
